src/coreclr/src/jit/assertionprop.cpp

```diff
@@ -4947,6 +4955,12 @@ GenTree* Compiler::optVNConstantPropOnJTrue(BasicBlock* block, GenTree* test)
 //
 Compiler::fgWalkResult Compiler::optVNConstantPropCurStmt(BasicBlock* block, Statement* stmt, GenTree* tree)
 {
+    // Don't perform const prop on expressions marked with GTF_DONT_CSE
+    if (!tree->CanCSE())
```

Comment:
was it unprofitable or incorrect to do?

Reply:
Doing a constant prop here would replace the CSE LclVar with the original constant.
Essentially undoing the CSE of the constant.